src/main/java/io/quarkus/gizmo2/impl/AnonymousClassCreatorImpl.java

```diff
@@ -32,6 +36,7 @@ public final class AnonymousClassCreatorImpl extends ClassCreatorImpl implements
     AnonymousClassCreatorImpl(final ClassDesc type, final ClassOutput output, final ClassBuilder zb,
             final ConstructorDesc superCtor, final ArrayList<Expr> captureExprs) {
         super(type, output, zb);
+        flags |= ACC_FINAL;
```

Comment:
I see this was removed from `BlockCreatorImpl.lambda()` and added here, but this also affects `BlockCreator.newAnonymousClass()`, which maybe isn't expected?

Reply:
Good point.